Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerSystemComponent.cpp

```diff
@@ -74,14 +76,21 @@ namespace AWSGameLift
 
     void AWSGameLiftServerSystemComponent::Activate()
     {
-        m_gameLiftServerManager->InitializeGameLiftServerSDK();
-        m_gameLiftServerManager->ActivateManager();
+        if (sv_useGameLiftServer)
+        {
+            m_gameLiftServerManager->InitializeGameLiftServerSDK();
+            m_gameLiftServerManager->ActivateManager();
+        }
+        
     }
 
     void AWSGameLiftServerSystemComponent::Deactivate()
     {
-        m_gameLiftServerManager->DeactivateManager();
-        m_gameLiftServerManager->HandleDestroySession();
+        if (sv_useGameLiftServer)
+        {
+            m_gameLiftServerManager->DeactivateManager();
+            m_gameLiftServerManager->HandleDestroySession();
+        }

```

Comment:
Because there's nothing preventing a user from trying to change this cvar after the system component has activated, the deactivate should probably be based on whether or not the m_gameLiftServerManager is actually active, not on the value of the cvar.

Reply:
How about just checking if m_gameLiftServerManager is valid and is active?

Reply:
`m_gameLiftServerManager` gets instantiated as a member var, so AFAICT it'll always exist ...unless that should change, and we should only instantiate it if `sv_useGameLiftServer`? Then after activation, we could check whether it exists to determine shut down behavior instead of adding a dedicate "active" status variable